### DISCUSSION TITLE
got 503 error response from mirth server

### DISCUSSION THREAD

**Discussion by partha1998** [nextgenhealthcare/connect#6380] [Q&A]
i have created a channel with http listener 
while i call the end point of http listener it worked from localhost with javascript axios package also worked in postman but not working on development server with axios package , it gives 503 error status code which is service unavailable . could anyone help me on this issue ?

**Comment by pacmano1** [comment]
I already commented on your forums post.    If you post here and the forums, please reference that fact.

You are providing a "problem" with little background, screenshots, code, etc.   Please read the forum etiquette on how to ask for help.

A 503 can occur for a number of reasons, so you need to provide more information.

**Reply by partha1998** [reply]
Thank you for the information 
yes this is the same issue what i have post it in the forum 
https://forums.mirthproject.io/forum/mirth-connect/support/185941-while-calling-http-listener-on-mirth-server-got-503-error-response
Now onwards we can continue over here with this issue 

This is the overal code in javascript :

import axiosRaw from 'axios'
  (async () => {
    const axiosResponse = await axiosRaw.post(mirth_url,data);
  })()
  
  OS : Ubuntu
  java version : 11
  mirth version : 4.5.2
  
  The error response is this : 
  response: {
    status: 503,
    statusText: 'Service Unavailable',
    headers: Object [AxiosHeaders] {
      'content-length': '91',
      'content-type': 'text/plain',
      date: 'Tue, 17 Dec 2024 13:53:28 GMT',
      server: 'envoy',
      connection: 'close'
    },
    config: {
      transitional: [Object],
      adapter: [Array],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 0,
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      env: [Object],
      validateStatus: [Function: validateStatus],
      headers: [Object [AxiosHeaders]],
      method: 'post',
      url: '<channel_http_listener_url>',
      data: '{"url":<url>}'
    },
    request: <ref *1> ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      maxRequestsOnConnectionReached: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      strictContentLength: false,
      _contentLength: '295',
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      _closed: false,
      socket: [Socket],
      _header: 'POST /api/getdata/ HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: application/json\r\n' +
        'User-Agent: axios/1.6.8\r\n' +
        'Content-Length: 295\r\n' +
        'Accept-Encoding: gzip, compress, deflate, br\r\n' +
        'Host: mirth-staging.sonora.telemed2u.com\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: nop],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      joinDuplicateHeaders: undefined,
      path: '/api/getdata/',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'mirth-staging.sonora.telemed2u.com',
      protocol: 'http:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kBytesWritten)]: 0,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype],
      [Symbol(errored)]: null,
      [Symbol(kHighWaterMark)]: 16384,
      [Symbol(kRejectNonStandardBodyWrites)]: false,
      [Symbol(kUniqueHeaders)]: null
    },
    data: 'upstream connect error or disconnect/reset before headers. reset reason: connection timeout'
  }
